swift/ios-shoppe-demo/ServiceManagers/AppStateController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  AppStateController.swift
+//  ios-shoppe-demo
+//
+//  Created by Harold Davis Jr. on 3/30/20.
+//  Copyright © 2020 Harold Davis Jr. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class AppStateController {
+    static var sharedInstance = AppStateController()
+
+    var user: String?
+    var usernameTextField: UITextField?
+    // TODO: var currentOrder: Order
+    // TODO: func addToCart(product: Product)
+
+    func displayAlert(view: UIViewController, completion: @escaping()->()) {
+        let createUserAlert = UIAlertController(title: "Sign in to to start shoping.", message: nil, preferredStyle: .alert)
+        createUserAlert.addTextField { (textField) in
+            self.usernameTextField = textField
+        }
+
+        let okAction = UIAlertAction(title: "Ok", style: .default) { (_) in
+            self.confirmUser()
+            completion()
+        }
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (_) in
+            completion()
+        }
+
+        createUserAlert.addAction(okAction)
+        createUserAlert.addAction(cancelAction)
+
+        view.present(createUserAlert, animated: true)
+    }
+
+    func userField(textField: UITextField!) {
```

Comment:
Is this code used?

Reply:
no it is not I actually would like to approach this another way, i was rushing to much to get this done. Im going to create an OrderManager instead of this AppState Controller and I will add the alert to the first ViewController for now.